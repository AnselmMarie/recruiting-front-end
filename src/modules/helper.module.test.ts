/* Module */
import transformNavJSON from './helper.module';
/* Type */
import testData from '../data/testData.json';

test('transform the json into a useable navigation', () => {
  const results = transformNavJSON(testData);
  expect(results).toMatchSnapshot();
});
