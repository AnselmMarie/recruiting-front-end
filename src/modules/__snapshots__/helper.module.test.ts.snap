// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform the json into a useable navigation 1`] = `
Array [
  Object {
    "child": Array [
      Object {
        "child": Array [
          Object {
            "id": 3,
            "name": "Node 3",
            "parent": 1,
            "thumbnail": Object {
              "description": "Picture, Random",
              "href": "https://picsum.photos/200/200/?random&d",
            },
          },
        ],
        "id": 1,
        "name": "Node 1",
        "parent": 0,
        "thumbnail": Object {
          "description": "Another Random Picture",
          "href": "https://picsum.photos/200/200/?random&b",
        },
      },
    ],
    "id": 0,
    "name": "Node 0",
    "parent": null,
    "thumbnail": Object {
      "description": "Random Picture",
      "href": "https://picsum.photos/200/200/?random&a",
    },
  },
  Object {
    "id": 2,
    "name": "Node 2",
    "parent": null,
    "thumbnail": Object {
      "description": "A Picture Is Random",
      "href": "https://picsum.photos/200/200/?random&c",
    },
  },
  Object {
    "child": Array [
      Object {
        "id": 4,
        "name": "Node 4",
        "parent": 5,
        "thumbnail": Object {
          "description": "rand(pix)",
          "href": "https://picsum.photos/200/200/?random&e",
        },
      },
    ],
    "id": 5,
    "name": "Node 5",
    "parent": null,
    "thumbnail": Object {
      "description": "pickPix <- rand",
      "href": "https://picsum.photos/200/200/?random&f",
    },
  },
]
`;
